crud/count.lua

```diff
@@ -0,0 +1,213 @@
+        count_to_yield = '?number',
+        scan_condition_num = '?number',
+    })
+
+    opts = opts or {}
+
+    local space = box.space[space_name]
+
+    local index = space.index[index_id]
+    if index == nil then
+        return nil, CountError:new("Index with ID %s doesn't exist", index_id)
+    end
+
+    local value = opts.scan_value
+
+    local filter_func, err = filters.gen_func(space, conditions, {
+        tarantool_iter = opts.tarantool_iter,
+        scan_condition_num = opts.scan_condition_num,
+    })
+    if err ~= nil then
+        return nil, CountError:new("Failed to generate tuples filter: %s", err)
+    end
+
+    local tuples_count = 0
+
+    local tuple
+    local gen = index:pairs(value, {iterator = opts.tarantool_iter})
+
+    gen.state, tuple = gen(gen.param, gen.state)
+    while true do
+        if tuple == nil then
+            break
+        end
+
+        local matched, early_exit = filter_func(tuple)
+
+        if matched then
+            tuples_count = tuples_count + 1
+
+            if opts.count_to_yield ~= nil and tuples_count % opts.count_to_yield == 0 then
+                fiber.yield()
+            end
+        elseif early_exit then
+            break
+        end
+
+        gen.state, tuple = gen(gen.param, gen.state)
+    end
+
+    return tuples_count
+end
+
+function count.init()
+    _G._crud.count_on_storage = count_on_storage
+end
+
+--- Calculates the number of tuples by conditions
+--
+-- @function call
+--
+-- @param string space_name
+--  A space name
+--
+-- @param ?table user_conditions
+--  Conditions by which tuples are counted,
+--  default value is empty table
+--
+-- @tparam ?number opts.timeout
+--  Function call timeout, default value is 2
+--
+-- @tparam ?number opts.count_to_yield
+--  Number of tuples processed to yield after,
+--  default value is 100
+--
+-- @tparam ?boolean opts.prefer_replica
+--  Call on replica if it's possible,
+--  default value is `nil`, which works as with `false`
+--
+-- @tparam ?boolean opts.balance
+--  Use replica according to round-robin load balancing
+--  default value is `nil`, which works as with `false`
+--
+-- @tparam ?string opts.mode
+--  vshard call mode, default value is `read`
+--
+-- @return[1] number
+-- @treturn[2] nil
+-- @treturn[2] table Error description
+--
+function count.call(space_name, user_conditions, opts)
+    checks('string', '?table', {
+        timeout = '?number',
+        bucket_id = '?number|cdata',
+        count_to_yield = '?number',
+        prefer_replica = '?boolean',
+        balance = '?boolean',
+        mode = '?string',
+    })
+
+    opts = opts or {}
+
+    if opts.count_to_yield ~= nil and opts.count_to_yield < 1 then
+        return nil, CountError:new("count_to_yield should be > 0")
+    end
+
+    -- check conditions
+    local conditions, err = compare_conditions.parse(user_conditions)
+    if err ~= nil then
+        return nil, CountError:new("Failed to parse conditions: %s", err)
+    end
```

Comment:
ldoc comment for `count.call` says that the default value for option `user_conditions` is an empty table.
If `user_conditions` is not specified, it value is `nil`. Then we pass user_conditions to `compare_conditions.parse()` and it returns `nil`. And then you pass `conditions` equal to `nil` to `count_plan.new()`. Seems something is wrong in ldoc comment or in a code. Default value definitely is not an empty table.

Reply:
ldoc comment was updated:

```
--  @param ?table user_conditions
--  Conditions by which tuples are counted,
--  default value is nil
```